sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Listeners/ServiceBusCausalityHelper.cs

```diff
@@ -2,23 +2,41 @@
 // Licensed under the MIT License. See License.txt in the project root for license information.
 
 using System;
-using Microsoft.Azure.ServiceBus;
+using Azure.Messaging.ServiceBus;
 
 namespace Microsoft.Azure.WebJobs.ServiceBus.Listeners
 {
     internal static class ServiceBusCausalityHelper
     {
         private const string ParentGuidFieldName = "$AzureWebJobsParentId";
 
-        public static void EncodePayload(Guid functionOwner, Message msg)
+        public static void EncodePayload(Guid functionOwner, ServiceBusMessage msg)
         {
-            msg.UserProperties[ParentGuidFieldName] = functionOwner.ToString();
+            msg.ApplicationProperties[ParentGuidFieldName] = functionOwner.ToString();
         }
 
-        public static Guid? GetOwner(Message msg)
+        public static Guid? GetOwner(ServiceBusReceivedMessage msg)
         {
             object parent;
-            if (msg.UserProperties.TryGetValue(ParentGuidFieldName, out parent))
+            if (msg.ApplicationProperties.TryGetValue(ParentGuidFieldName, out parent))
+            {
+                var parentString = parent as string;
+                if (parentString != null)
+                {
+                    Guid parentGuid;
+                    if (Guid.TryParse(parentString, out parentGuid))
+                    {
+                        return parentGuid;
+                    }
+                }
+            }
```

Comment:
Or even 
```
if (msg.ApplicationProperties.TryGetValue(ParentGuidFieldName, out var parent) &&
    Guid.TryParse(parent?.ToString(), out var parentGuid))
{
    return parentGuid;
}

return null;
```

Reply:
since we are tossing out code snippets, how about:
```cs
        public static Guid? GetOwner(IDictionary<string, object> properties)
        {
            if (properties.TryGetValue(ParentGuidFieldName, out var parent))
            {
                if (Guid.TryParse(parent?.ToString(), out var parentGuid))
                {
                    return parentGuid;
                }
            }
            return null;
        }
```
to avoid repeating for both types of message.

Reply:
Actually nevermind, the other one wasn't used.